app/core/core/src/main/java/io/syndesis/core/KeyGenerator.java

```diff
@@ -76,4 +79,39 @@ public static String createKey() {
             return randomVal;
         });
     }
+
+    /**
+     * Used to extract the time information that is encoded to each
+     * generated key.
+     */
+    public static long getKeyTimeMillis(String key) throws IOException {
```

Comment:
a feaure only needed for the logging, not for creating DB keys.(yet another smell that these should be two classes).

Reply:
"two classes' of KeyGenerator?

Reply:
All these generated keys have had an embedded timestamp component.  The keys have not changed. We are just adding methods that further expose USEFUL information.  When statements like these are made could you please also state scenarios where doing these things harms something plz.

Reply:
I said it's a _smell_, not an issue. You didn't need the timestamp for the original usage, but you do need it now. Maybe because its so useful to embedd semantic information into a key, maybe you drink the Kool-Aid and want to add more information like this to the key later. But changing this class on behalf of one of the dependants, requires to update the other dependent, too. Or to be extra careful to not break stuff (if we still remember that there are other dependents).

Not an issue now, just a smell. But why adding this smell when it is so easy to avoid it in the first place ? That's my opinion (i.e. that runtime code should not share code with backend code, as long as we don't hit a hard wall). Separation of concerns is a commonly accepted principle. 

Don't want to argue further, as said previously, I'm somewhat ok with that.